File: samples/10_modular-kmp-data-layer/androidApp/src/main/kotlin/com/example/kmpsamples/modularkmpdatalayer/android/DetailsViewModel.kt
package com.example.kmpsamples.modularkmpdatalayer.android

import com.example.kmpsamples.modularkmpdatalayer.featuretwoshareddata.DetailsRepository

class DetailsViewModel(
    private val repository: DetailsRepository = DetailsRepository()
) {
    fun screenState(platform: String): String = "Native DetailsViewModel -> ${repository.loadDetailsSummary(platform)}"
}

File: samples/09_modular-native/iosApp/Sources/DetailsViewModel.swift
final class DetailsViewModel {
    private let repository: DetailsRepository

    init(repository: DetailsRepository = DetailsRepository(dataSource: DetailsDataSource())) {
        self.repository = repository
    }

    func screenState(platform: String) -> String {
        repository.loadDetailsSummary(platform: platform)
    }
}
